oidc/provider.go

```diff
@@ -306,6 +307,12 @@ func (p *Provider) Exchange(ctx context.Context, oidcRequest Request, authorizat
 	if oidcRequest.PKCEVerifier() != nil {
 		authCodeOpts = append(authCodeOpts, oauth2.SetAuthURLParam("code_verifier", oidcRequest.PKCEVerifier().Verifier()))
 	}
+	if oidcRequest.ClientAssertionJWT() != "" {
+		authCodeOpts = append(authCodeOpts,
+			oauth2.SetAuthURLParam("client_assertion_type", cass.ClientAssertionJWTType),
+			oauth2.SetAuthURLParam("client_assertion", oidcRequest.ClientAssertionJWT()),
+		)
+	}
```

Comment:
Is this exclusive with any other option? I'm not too familiar with this OAuth2 pattern.

Reply:
I know that it at least plays nicely with PKCE. If the assertion is signed by the ClientSecret, then it sorta re-contextualizes that param, but otherwise I don't think it's exclusive of other things? Perhaps I'm ignorant of some nuance here though (there is a lot of nuance around OIDC, I've learned).

Reply:
I found this particular implementation pattern, using `oauth2.SetAuthURLParam`, in this issue where someone requested it in the `oauth2` repo here: https://github.com/golang/oauth2/issues/744#issuecomment-2411878895